.github/workflows/test.yml

```diff
@@ -0,0 +1,245 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: '0 6 * * *'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    # Use an interactive shell so that ~/.bashrc is sourced:
+    shell: 'bash -eio pipefail {0}'
+
+env:
+   # Note, that there's "turtle" as well, which is always excluded from running.
+   DEFAULT_PYTEST_SELECTION: "integration or usecase or slow or network"
+   DEFAULT_PYTEST_SELECTION_OP: "not "  # so it would be "not (integration or usecase)"
+   DEFAULT_DATALAD_TESTS_SSH: "1"
+   DEFAULT_DATALAD_LOG_ENV: GIT_SSH_COMMAND
```

Comment:
why do we need `DEFAULT_{VARIABLE}` construct and can't we just set default values here and overload in the matrix???

Reply:
I'm not 100% sure if that would work.  The envvar setup went through several rounds of more complicated configurations before it got where it is now, so some simplifications may have been overlooked.

Reply:
please check and simplify if possible.  I hope I am done fixing up those matrix runs (although yet to see the final run now in a cleaned up state)

Reply:
a2f9cf527 should work.